functional/keylime_tenant-ima-signature-sanity/test.sh

```diff
@@ -0,0 +1,121 @@
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Test that IMA key verification using GPG signature fails for an invalid key"
+        rlRun "gpg --verify signature-gpg-fake.sig allowlist.txt"
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --sign_verification_key ${limeIMAPublicKey} --signature-verification-key-sig signature-gpg-fake.sig --signature-verification-key-sig-key gpg-key.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartTest "Verify IMA key using a downloaded GPG signature and key"
+        rlRun "cp ${limeIMAPublicKey} ."
+        rlRun "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --signature-verification-key-url 'http://localhost:8000/x509_evm.pem' --signature-verification-key-sig-url 'http://localhost:8000/signature-gpg-genuine.sig' --signature-verification-key-sig-url-key gpg-key.pub -c update"
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Test that IMA key verification using downloaded GPG signature fails for an invalid key"
+        rlRun "cp ${limeIMAPublicKey} ."
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --signature-verification-key-url 'http://localhost:8000/x509_evm.pem' --signature-verification-key-sig-url 'http://localhost:8000/signature-gpg-fake.sig' --signature-verification-key-sig-url-key gpg-key.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartCleanup "Do the keylime cleanup"
+        rlRun "kill $HTTP_PID"
+        rlRun "limeStopAgent"
+        rlRun "limeStopRegistrar"
+        rlRun "limeStopVerifier"
+        limeLogfileSubmit $(limeVerifierLogfile)
+        limeLogfileSubmit $(limeRegistrarLogfile)
+        limeLogfileSubmit $(limeAgentLogfile)
+        if limeTPMEmulated; then
+            rlRun "limeStopIMAEmulator"
+            limeLogfileSubmit $(limeIMAEmulatorLogfile)
+            rlRun "limeStopTPMEmulator"
+            rlServiceRestore tpm2-abrmd
+        fi
+        limeClearData
+        limeRestoreConfig
+        limeExtendNextExcludelist ${TESTDIR}
+        rlRun "rm -rf ${TmpDir}"
+        rlFileRestore
+        rlRun "gpgconf --kill gpg-agent"
```

Comment:
Just a nitpick but better stop `gpg-agent` first and then call `rlFileRestore`, just to make sure agent won't save some data there while stopping.